projects/graphql-client/src/utils/builders/request/graphql-request-builder.ts

```diff
@@ -132,7 +132,7 @@ export class GraphQlRequestBuilder {
       return `[${arrayContents}]`;
     }
 
-    if (typeof argumentValue === 'object' && !(argumentValue instanceof Date)) {
+    if (typeof argumentValue === 'object' && !isNil(argumentValue) && !(argumentValue instanceof Date)) {
```

Comment:
See my earlier suggestion offline - we should translate any nil argument as null. this would handle null correctly, but undefined would not be converted to null.

Reply:
done